server/controllers/configurationsController/configurationsController.test.js

```diff
@@ -0,0 +1,175 @@
+import ConfigurationsController from '.'
+import { createRequestWithUser, createResponse } from '../../../test/fixtures'
+
+describe('ConfigurationsController', () => {
+  describe(ConfigurationsController.create, () => {
+    const newConfiguration = { body: { name: 'new matrix', owner: 'owl' } }
+    const request = createRequestWithUser(newConfiguration)
+    const response = createResponse()
+
+    describe('When successful', () => {
+      it('sends a status of 200 with a data property', async () => {
+        request.app.locals.appDb.insertOne.mockResolvedValueOnce({
+          insertedId: 'eagle',
+        })
+
+        await ConfigurationsController.create(request, response)
+
+        expect(response.status).toHaveBeenCalledWith(200)
+        expect(response.json).toHaveBeenCalledWith({
+          data: { insertedId: 'eagle' },
+        })
+      })
+    })
+
+    describe('when unsuccessful', () => {
+      it('returns a 400 error with an error message', async () => {
+        request.app.locals.appDb.insertOne.mockImplementation(() => {
+          new Error('some error')
+        })
+
+        await ConfigurationsController.create(request, response)
+
+        expect(response.status).toHaveBeenCalledWith(400)
+        expect(response.json).toHaveBeenCalledWith({
+          error: new Error('some error'),
+        })
+      })
+    })
+  })
+
+  describe(ConfigurationsController.edit, () => {
+    const configAttributes = { body: { name: 'new matrix', owner: 'owl' } }
+    const request = createRequestWithUser(configAttributes)
+    const response = createResponse()
+
+    describe('When successful', () => {
+      it('sends a status of 200 and with a data payload', async () => {
+        request.app.locals.appDb.findOneAndUpdate.mockResolvedValueOnce({
+          modifiedCount: 1,
+        })
+
+        await ConfigurationsController.edit(request, response)
+
+        expect(response.status).toHaveBeenCalledWith(200)
+        expect(response.json).toHaveBeenCalledWith({
+          modifiedCount: 1,
+        })
```

Comment:
can we return the updated configuration in the response instead of the modified count?

Reply:
[f11e811](https://github.com/AMP-SCZ/dpdash/pull/305/commits/f11e811a37bb7df8cc158a8d0e061615e8ec535d)